test/gurka/test_matrix.cc

```diff
@@ -87,11 +87,13 @@ class MatrixTest : public ::testing::Test {
                               {"KL", {{"highway", "primary"}}}};
 
     const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize);
+    // also turn on the reverse connection search; there's no real test for it
     map = gurka::buildtiles(layout, ways, {}, {}, "test/data/matrix_traffic_allowed",
                             {{"service_limits.max_timedep_distance_matrix", "50000"},
                              {"mjolnir.traffic_extract",
                               "test/data/matrix_traffic_allowed/traffic.tar"},
-                             {"mjolnir.timezone", "test/data/tz.sqlite"}});
+                             {"mjolnir.timezone", "test/data/tz.sqlite"},
+                             {"thor.costmatrix_check_reverse_connection", "1"}});
```

Comment:
updated this test to turn on the new option. it'd be pretty tedious to come up with a test that shows the benefit of it. the request at the original issue in https://github.com/valhalla/valhalla/issues/4274 works now though